packages/gatsby/src/services/initialize.ts

```diff
@@ -659,10 +659,12 @@ export async function initialize({
 
   const workerPool = WorkerPool.create()
 
-  // This is only run during `gatsby develop`
   if (state.config.graphqlTypegen) {
     telemetry.trackFeatureIsUsed(`GraphQLTypegen`)
-    writeGraphQLConfig(program)
+    // This is only run during `gatsby develop`
```

Comment:
This is correcting the implementation to actually do what the comment said (this file is not needed during gatsby build)